pyccel/codegen/printing/cwrappercode.py

```diff
@@ -57,49 +60,51 @@ def _print_FunctionDef(self, expr):
             if cast_func is not None:
                 # TODO: Add other properties
                 collect_var = Variable(dtype=collect_type,
-                        name=get_new_name(used_names, a.name+"_tmp"))
+                        name=self.et_new_name(used_names, a.name+"_tmp"))
                 wrapper_vars.append(collect_var)
                 parse_args.append(collect_var)
                 wrapper_body.append(cast_func(a, collect_var))
             else:
                 parse_args.append(a)
-            type_keys.append(pytype_registry[str_dtype(arg.dtype)])
+            type_keys += pytype_registry[a.dtype]
 
         # TODO: Create PyArg_ParseTupleNode
-        wrapper_body.insert(0,If((Not(PyArg_ParseTupleNode(python_func_args, type_keys, parse_args)), Return(Nil))))
+        wrapper_body.insert(0,If((PyccelNot(PyArg_ParseTupleNode(python_func_args, type_keys, parse_args)), [Return([Nil])])))
 
         if len(expr.results)==0:
             func_call = FunctionCall(expr, expr.arguments)
         else:
             results = expr.results if len(expr.results)>1 else expr.results[0]
             func_call = Assign(results,FunctionCall(expr, expr.arguments))
         wrapper_body.append(func_call)
-
         #TODO: Loop over results to carry out necessary casts and collect Py_BuildValue type string
-
+        name = expr.name.name+"_wrapper"
```

Comment:
We should use get_new_name in case of name collisions

Reply:
ok i'll do it